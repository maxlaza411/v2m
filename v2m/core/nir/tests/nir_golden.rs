use std::path::PathBuf;

use v2m_formats::load_nir;
use v2m_nir::normalize_nir;

fn data_dir() -> PathBuf {
    PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("../../tests/golden/nir")
}

fn load_normalized(name: &str) -> v2m_nir::NormalizedNir {
    let path = data_dir().join(format!("{name}.nir.json"));
    let nir = load_nir(&path).expect("load nir design");
    normalize_nir(&nir).expect("normalize design")
}

fn assert_snapshot(name: &str) {
    let mut settings = insta::Settings::new();
    settings.set_snapshot_path(data_dir());
    settings.set_snapshot_suffix("snap.json");
    settings.set_omit_expression(true);
    settings.set_prepend_module_to_snapshot(false);
    let _guard = settings.bind_to_scope();

    let design = load_normalized(name);
    insta::assert_json_snapshot!(name, design);
}

#[test]
fn golden_fa1() {
    assert_snapshot("fa1");
}

#[test]
fn golden_alu4() {
    assert_snapshot("alu4");
}

#[test]
fn golden_reg1x8() {
    assert_snapshot("reg1x8");
}
