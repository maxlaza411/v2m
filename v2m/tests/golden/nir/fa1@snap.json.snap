---
source: core/nir/tests/nir_golden.rs
assertion_line: 25
---
{
  "design": "fa1",
  "top": "fa1",
  "modules": {
    "fa1": {
      "inputs": {
        "a": [
          {
            "node": 0
          }
        ],
        "b": [
          {
            "node": 1
          }
        ],
        "cin": [
          {
            "node": 2
          }
        ]
      },
      "outputs": {
        "cout": [
          {
            "node": 6
          }
        ],
        "sum": [
          {
            "node": 7
          }
        ]
      },
      "states": [],
      "nodes": [
        {
          "width": 1,
          "id": 0,
          "kind": {
            "kind": "input"
          }
        },
        {
          "width": 1,
          "id": 1,
          "kind": {
            "kind": "input"
          }
        },
        {
          "width": 1,
          "id": 2,
          "kind": {
            "kind": "input"
          }
        },
        {
          "width": 1,
          "id": 3,
          "kind": {
            "kind": "op",
            "op": "AND",
            "inputs": [
              {
                "node": 0
              },
              {
                "node": 1
              }
            ]
          }
        },
        {
          "width": 1,
          "id": 4,
          "kind": {
            "kind": "op",
            "op": "XOR",
            "inputs": [
              {
                "node": 0
              },
              {
                "node": 1
              }
            ]
          }
        },
        {
          "width": 1,
          "id": 5,
          "kind": {
            "kind": "op",
            "op": "AND",
            "inputs": [
              {
                "node": 2
              },
              {
                "node": 4
              }
            ]
          }
        },
        {
          "width": 1,
          "id": 6,
          "kind": {
            "kind": "op",
            "op": "OR",
            "inputs": [
              {
                "node": 3
              },
              {
                "node": 5
              }
            ]
          }
        },
        {
          "width": 1,
          "id": 7,
          "kind": {
            "kind": "op",
            "op": "XOR",
            "inputs": [
              {
                "node": 2
              },
              {
                "node": 4
              }
            ]
          }
        }
      ]
    }
  }
}
